src/lib/input/input-container.ts

```diff
@@ -215,10 +224,14 @@ export class MdInputDirective {
               private _renderer: Renderer2,
               @Optional() @Self() public _ngControl: NgControl,
               @Optional() private _parentForm: NgForm,
-              @Optional() private _parentFormGroup: FormGroupDirective) {
+              @Optional() private _parentFormGroup: FormGroupDirective,
+              @Optional() @Inject(MD_ERROR_GLOBAL_OPTIONS) errorOptions: ErrorOptions) {
 
     // Force setter to be called in case id was not specified.
     this.id = this.id;
+
+    this._errorOptions = errorOptions ? errorOptions : {};
+    this.errorStateMatcher = this._errorOptions.errorStateMatcher || undefined;
```

Comment:
if you create the `DefaultErrorStateMatcher` class I suggested earlier this can be `|| new DefaultErrorStateMatcher()`